modules/web/web-api/src/main/java/com/enonic/xp/web/servlet/ServletRequestUrlHelper.java

```diff
@@ -134,7 +138,7 @@ public static String getServerUrl( final HttpServletRequest req )
         str.append( scheme );
 
         //Appends the host
-        str.append( "://" ).append( req.getServerName() );
+        str.append( "://" ).append( Objects.requireNonNullElse( getCustomHostFromContext(), req.getServerName() ) );
```

Comment:
Configuring host is not sufficient. It should be baseUrl.
Example: `https://example.com/some/path`